src/java.base/share/classes/java/io/DataOutputStream.java

```diff
@@ -37,6 +37,11 @@
  * thread then access to the data output stream should be controlled
  * by appropriate synchronization.
  *
+ * @apiNote
+ * The write methods of this class merely delegate to the methods of
+ * the underlying output stream, hence there is no guarantee as to
+ * the details of how the bytes are written.
+ *
```

Comment:
I'd drop "merely".

I would not want to give any assurances about whether the bytes of an encoded primitive were written individually or as a group.  
For the primitive types the value is encoded and written as one or more bytes to the underlying stream.

Reply:
> I would not want to give any assurances about whether the bytes of an encoded primitive were written individually or as a group.

Would it be helpful to add further verbiage?
```
For example, the sequence of bytes comprising the primitive data type may be written as more than one packet of bytes.
```

Reply:
"packet" is network centric terminology.
Perhaps something like:  "may be written individually or in any grouping of bytes"

Reply:
Commit 43c392e8c797f5a5c29a67e87f350e8dc9c5b190 changed it to
```
The write methods of this class delegate to the methods
of the underlying output stream, hence there is no 
guarantee as to the details of how the bytes are written.
```